src/main/kotlin/org/jetbrains/kotlinx/jupyter/apiImpl.kt

```diff
@@ -132,6 +136,23 @@ class NotebookImpl(
     override val jreInfo: JREInfoProvider
         get() = JavaRuntime
 
+    fun updateVarsState(varsMap: Map<String, VariableState>) {
+        variablesMap += varsMap
+    }
+
+    fun varsAsHtmlTable(): String {
+        return generateHTMLVarsReport(variablesMap)
+    }
+
+    fun varsAsString(): String {
+        if (variablesMap.isEmpty()) return ""
+        val stringBuilder = StringBuilder("Visible vars: \n")
+        variablesMap.forEach { (t, u) ->
+            stringBuilder.append("\t$t : ${u.stringValue}\n")
+        }
+        return stringBuilder.append('\n').toString()
```

Comment:
Use `buildString` to save a couple of characters. You may use saved characters for specifying understandable names for `t` and in `u` in `forEach`